resources/sdk/pureclouddotnet/templates/ApiClient.mustache

```diff
@@ -275,6 +275,29 @@ namespace {{packageName}}.Client
                 return Convert.ToString (obj);
         }
 
+         /// <summary>
+        /// Creates a restclient with a base path string input
+        /// <returns>Returns a rest client</returns>
+        public RestClient withBasePath(String basePath){
+            if (String.IsNullOrEmpty(basePath))
+                throw new ArgumentException("basePath cannot be empty");
+
+                 RestClient = new RestClient(basePath);
```

Comment:
Instead of creating a new RestClient instance, just set the value: `RestClient.BaseUrl = new Uri(basePath);`